shotover-proxy/src/transforms/cassandra/sink_cluster/mod.rs

```diff
@@ -604,25 +604,37 @@ impl CassandraSinkCluster {
         ))
     }
 
+    /// Returns any information required to correctly rewrite the response.
+    /// Will also perform minor modifications to the query required for the rewrite.
     fn get_rewrite_table(&self, request: &mut Message, index: usize) -> Option<TableToRewrite> {
         if let Some(Frame::Cassandra(cassandra)) = request.frame() {
             // No need to handle Batch as selects can only occur on Query
-            if let CassandraOperation::Query { query, .. } = &cassandra.operation {
-                if let CassandraStatement::Select(select) = query.as_ref() {
+            if let CassandraOperation::Query { query, .. } = &mut cassandra.operation {
+                if let CassandraStatement::Select(select) = query.as_mut() {
                     let ty = if self.local_table == select.table_name {
                         RewriteTableTy::Local
                     } else if self.peers_table == select.table_name
                         || self.peers_v2_table == select.table_name
                     {
-                        // TODO: fail if WHERE exists
                         RewriteTableTy::Peers
                     } else {
                         return None;
                     };
 
+                    let warnings = if Self::has_no_where_clause(ty, select) {
+                        vec![]
+                    } else {
+                        select.where_clause.clear();
+                        vec![format!(
+                            "WHERE clause on the query was ignored. Shotover does not support WHERE clauses on queries against {}",
```

Comment:
Do we know of any concrete situations (specific drivers) where this can be triggered for `system.peers`? Is it more important to try and return something, versus failing early like the TODO said, to minimise the risk of incorrect behaviour?

Reply:
I do not know of any concrete instances where this can be triggered for system.peers.

ben did voice the opinion that we would be better off returning everything and hope the driver gets what it needs: https://github.com/shotover/shotover-proxy/pull/779#issuecomment-1233697939

Personally I would return an error here, but I suspect that I tend to favor correctness over compatibility a bit too much.